src/sentry/static/sentry/app/components/search/searchResult.jsx

```diff
@@ -0,0 +1,128 @@
+import {Flex} from 'grid-emotion';
+import {withRouter} from 'react-router';
+import PropTypes from 'prop-types';
+import React from 'react';
+import styled from 'react-emotion';
+
+import Avatar from '../avatar';
+import InlineSvg from '../inlineSvg';
+import SentryTypes from '../../proptypes';
+import UserBadge from '../userBadge';
+import highlightFuseMatches from '../../utils/highlightFuseMatches';
+
+class SearchResult extends React.Component {
+  static propTypes = {
+    item: PropTypes.shape({
+      /**
+     * The source of the search result (i.e. a model type)
+     */
+      sourceType: PropTypes.oneOf([
+        'organization',
+        'project',
+        'team',
+        'member',
+        'field',
+        'route',
+      ]),
+      /**
+     * The type of result this is, for example:
+     * - can be a setting route,
+     * - an application route (e.g. org dashboard)
+     * - form field
+     */
+      resultType: PropTypes.oneOf(['settings', 'route', 'field']),
+      title: PropTypes.string,
+      description: PropTypes.string,
+      model: PropTypes.oneOfType([
+        SentryTypes.Organization,
+        SentryTypes.Project,
+        SentryTypes.Team,
+        SentryTypes.Member,
+      ]),
+    }),
+    matches: PropTypes.array,
+  };
+
+  renderContent() {
+    let {item, matches, params} = this.props;
+    let {sourceType, title, description, model} = item;
+
+    let matchedTitle = matches && matches.find(({key}) => key === 'title');
+    let matchedDescription = matches && matches.find(({key}) => key === 'description');
+    let highlightedTitle = matchedTitle ? highlightFuseMatches(matchedTitle) : title;
+    let highlightedDescription = matchedDescription
+      ? highlightFuseMatches(matchedDescription)
+      : description;
+
+    if (sourceType === 'member') {
+      return (
+        <UserBadge
+          displayName={highlightedTitle}
+          displayEmail={highlightedDescription}
+          userLink={false}
+          orgId={params.orgId}
+          user={model}
+        />
+      );
+    }
+
+    return (
+      <React.Fragment>
+        <div>
+          {sourceType === 'team' && <TeamAvatar team={model} size={32} />}
```

Comment:
It's currently coupled because there are a few different ways to render based on `sourceType`. (User vs. Team vs. everything else).

The other method handles `resultType` which currently is only used for the icon on the right (subject to change based on actual usefulness)